hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/token/delegation/ZKDelegationTokenSecretManager.java

```diff
@@ -417,24 +408,18 @@ private void processTokenAddOrUpdate(byte[] data) throws IOException {
     if (numRead > -1) {
       DelegationTokenInformation tokenInfo =
           new DelegationTokenInformation(renewDate, password);
-      synchronized (this) {
```

Comment:
This change removes a few synchronized keywords. I cross-check with trunk and trunk also removes these. seems fine but I have no context on why synchronized was used before and now removed.

Reply:
It was introduced from https://github.com/apache/hadoop/pull/2047/files 
I don't get the full context of that change but in `AbstractDelegationTokenSecretManager.java`, they changed implementations of `currentTokens` and `allKeys` from `hashmap` to `concurrentHashMap`(thread safe). That's why they removed `synchronized` as it's a duplicate protection.